src/macaron/repo_finder/commit_finder.py

```diff
@@ -473,42 +473,73 @@ def _compute_tag_version_similarity(tag_version: str, tag_suffix: str, version_p
     """
     count = len(version_parts)
     # Reduce count for each direct match between version parts and tag version.
-    tag_version_text = tag_version
+    tag_version_text = tag_version.lower()
     for part in version_parts:
+        part = part.lower()
         if part in tag_version_text:
             tag_version_text = tag_version_text.replace(part, "", 1)
             count = count - 1
 
     # Try to reduce the count further based on the tag suffix.
     if tag_suffix:
-        last_part = version_parts[-1]
+        last_part = version_parts[-1].lower()
         # The tag suffix might consist of multiple version parts, e.g. RC1.RELEASE
         suffix_split = split_pattern.split(tag_suffix)
+        # Try to match suffix parts to version.
+        versioned_string_match = False
         if len(suffix_split) > 1:
-            # Try to match suffix parts to version.
-            versioned_string_match = False
             for suffix_part in suffix_split:
+                suffix_part = suffix_part.lower()
                 if alphabetic_only_pattern.match(suffix_part) and suffix_part == last_part:
                     # If the suffix part only contains alphabetic characters, reduce the count if it
                     # matches the version.
                     count = count - 1
                     continue
-                if versioned_string.match(suffix_part):
-                    # If the suffix part contains alphabetic characters followed by numeric characters,
-                    # reduce the count if it matches the version (once only), otherwise increase the count.
-                    if not versioned_string_match and suffix_part == last_part:
-                        count = count - 1
-                        versioned_string_match = True
-                    else:
-                        count = count + 1
-        if tag_suffix != last_part:
-            count = count + 1
+
+                variable_suffix_pattern = _create_suffix_tag_comparison_pattern(suffix_part)
+                if not variable_suffix_pattern:
+                    continue
```

Comment:
Hmm why do we ignore the scenario where the suffix_part is in the form that we don't support (i.e. `_create_suffix_tag_comparison_pattern` returns None) Should we treat this tag as bad if it's malformed?

Reply:
If it returns None, that means the suffix is not an alphanumeric versioned string, and it's not an alphabetic string that matches the version part. In that case we do assume it is essentially junk. For a check that evaluates tags that could be an important thing to flag, but for these comparisons I don't think it is important.